Sources/SectionScrubber.swift

```diff
@@ -167,82 +196,89 @@ public class SectionScrubber: UIView {
     public func updateScrubberPosition() {
         guard let collectionView = self.collectionView else { return }
         guard collectionView.contentSize.height != 0 else { return }
-        guard let originalYOffset = self.originalYOffset else { return }
 
         self.userInteractionOnScrollViewDetected()
 
-        let initialY = self.containingViewFrame.height
-        let totalHeight = collectionView.contentSize.height - self.containingViewFrame.height
-        let currentY = (collectionView.contentOffset.y - originalYOffset) + self.containingViewFrame.height
-        let currentPercentage = (currentY - initialY) / totalHeight
-        let containerHeight = (self.containingViewFrame.height - self.viewHeight)
-        let y = (containerHeight * currentPercentage) + collectionView.contentOffset.y - originalYOffset
-        self.frame = CGRect(x: 0, y: y, width: collectionView.frame.width, height: self.viewHeight)
-
-        /**
-         Initial dragging doesn't take in account collection view headers, just cells, so before the scrubber reaches
-         a cell, this is not going to return an index path.
-        **/
-        let centerPoint = CGPoint(x: SectionScrubber.initialXCoordinateToCalculateIndexPath, y: self.center.y);
-        if let indexPath = collectionView.indexPathForItemAtPoint(centerPoint) {
+        let percentage = self.boundedPercentage(collectionView.contentOffset.y / self.adjustedContainerHeight)
+        let newY = self.adjustedContainerOffset + (self.adjustedContainerBoundsHeight * percentage)
+        self.topConstraint?.constant = newY
+
+        self.updateSectionTitle()
+    }
+
+    /**
+     Initial dragging doesn't take in account collection view headers, just cells, so before the scrubber reaches
+     a cell, this is not going to return an index path.
+     **/
+    private func indexPath(at point: CGPoint) -> NSIndexPath? {
+        if let indexPath = self.collectionView?.indexPathForItemAtPoint(point) {
+            return indexPath
+        }
+        return nil
+    }
+
+    private func updateSectionTitle() {
+        //WARNING: this makes too many assumptions about the collection view layout. 😔
```

Comment:
Remove the `WARNING` prefix, otherwise it will get displayed on Xcode of people using the pod.